eex/tests/test_datalayer.py

```diff
@@ -970,11 +970,85 @@ def test_remove_terms_by_index_nonconsecutive_propogate(butane_dl):
     assert (dl.get_term_count(3)['total'] == 2)
     assert (dl.get_term_count(4)['total'] == 1)
 
-    # Remove two bonds - choose to propogate this so dihedral and both angles should also be removed.
-    dl.remove_terms(2, index=[0, 2], propogate=True)
+    # Remove two bonds - choose to propagate this so dihedral and both angles should also be removed.
+    dl.remove_terms(2, index=[0, 2], propagate=True)
 
     assert(dl.get_term_count(2)['total'] == 1)
     assert (dl.get_term_count(3)['total'] == 0)
     assert(dl.get_term_count(4)['total'] == 0)
 
     return True
+
+def test_remove_and_add_terms(butane_dl):
+    dl = butane_dl()
+
+    bonds = dl.get_terms(2)
+
+    assert(not bonds.empty)
+
+    dl.remove_terms(2)
+
+    bonds_new = dl.get_terms(2)
+
+    assert(bonds_new.empty)
+    assert(dl.get_term_count(2)['total'] == 0)
+
+    dl.add_bonds(bonds)
+
+    added_bonds = dl.get_terms(2)
+
+    assert (not added_bonds.empty)
+
+def test_remove_and_add_terms_index(butane_dl):
+    dl = butane_dl()
+
+    bonds = dl.get_terms(2)
+
+    assert(not bonds.empty)
```

Comment:
Make sure to run `yapf` over these files. I also do not think you need parenthesis here.